go/worker/common/p2p/init.go

```diff
@@ -18,6 +18,9 @@ const (
 	CfgP2PPeerOutboundQueueSize = "worker.p2p.peer_outbound_queue_size"
 	// CfgP2PValidateQueueSize sets the libp2p gossipsub buffer size of the validate queue.
 	CfgP2PValidateQueueSize = "worker.p2p.validate_queue_size"
+	// CfgP2PConnectednessLowWater sets the ratio of connected to unconnected peers at which
+	// the peer manager will try to reconnect to disconnected nodes.
+	CfgP2PConnectednessLowWater = "worker.p2p.connectedness_low_water"
```

Comment:
Would it be too much to always try to reconnect to disconnected nodes?  If anything, a "per peer" back-off would make more sense if we feel trying to reconnect every 5 secs is to excessive.

Reply:
I changed this to a combination of both: the new code takes care to preserve the connect backoff that was already there for cases where the connect worker is already running (e.g. if there's a disconnected peer and the addresses remain unchanged), and there's a periodic sweep that triggers reconnecting on disconnected peers (which also implicitly handles the corner case where the connect worker was exiting for some reason and the peer still isn't connected).